### PR TITLE
Add support for NDK r19 prebuilt toolchains

### PR DESCRIPTION
**Pull request by MaulingMonkey**
zuurr ([on the community discord](https://discordapp.com/channels/273534239310479360/335502453371961344/619291453121757210)) suggested I migrate ncalexan/rust-android-gradle#36 against this repository.  This isn't my work originally, but I have tested it.  This could help with CI build times, and sidesteps the problem of `%LOCALAPPDATA%\Temp` being partially cleaned up on some machines.

Closes mozilla/rust-android-gradle#8
Closes ncalexan/rust-android-gradle#36

### CODE REVIEW

File: plugin/src/main/kotlin/com/nishtahir/CargoExtension.kt [hunk 1]
```diff
@@ -35,6 +35,7 @@ open class CargoExtension {
     var module: String? = null
     var libname: String? = null
     var targets: List<String>? = null
+    var prebuiltToolchains: Boolean? = null
```
Comment (thomcc) [inline plugin/src/main/kotlin/com/nishtahir/CargoExtension.kt]: So, I know you didn't write this initially, but I think we want to be able to configure this via local.properties as well. You can take a look at the other uses of localProperties in this file for reference. If it's specified in local.properties, I think that should take precedence.

File: plugin/src/main/kotlin/com/nishtahir/RustAndroidPlugin.kt [hunk 2]
```diff
@@ -20,67 +21,104 @@ val toolchains = listOf(
...
         Toolchain("x86_64",
-                ToolchainType.ANDROID,
+                ToolchainType.ANDROID_GENERATED,
+                "x86_64-linux-android",
+                "x86_64-linux-android",
+                "x86_64-linux-android",
+                "android/x86_64"),
+        Toolchain("arm",
+                ToolchainType.ANDROID_PREBUILT,
+                "armv7-linux-androideabi",
```
Comment (thomcc) [inline plugin/src/main/kotlin/com/nishtahir/RustAndroidPlugin.kt]: I know this isn't your work, but a comment mentioning that the differences here are intentional would probably not go amiss.

**Comment by MaulingMonkey** [comment]
> Would you mind fixing these?

Sure!

File: plugin/src/main/kotlin/com/nishtahir/CargoExtension.kt (hunk 1 above)
Comment (MaulingMonkey) [reply plugin/src/main/kotlin/com/nishtahir/CargoExtension.kt]: Implementing that logic directly in here is a bit awkward - the getter would need to effectively ignore the set value if overridden by localProperties.  Or I could have a separate overriddenPrebuiltToolchains, but that seems easy to overlook when I can't make prebuiltToolchains's getter private in Kotlin without making the setter private too.

RustAndroidPlugin::configurePlugin - the only user of prebuitlToolchains - already looks at localProperties too though, so I can implement local.properties override logic in there without things looking too awkward?

Comment (thomcc) [reply plugin/src/main/kotlin/com/nishtahir/CargoExtension.kt]: > RustAndroidPlugin::configurePlugin - the only user of prebuitlToolchains - already looks at localProperties too though, so I can implement local.properties override logic in there without things looking too awkward?

SGTM, I didn't mean to imply that it needed to be done in the place where I commented, it was just a convenient place to mention it.

**Comment by MaulingMonkey** [comment]
Believe I've addressed everything, let me know if there's anything I've missed or more stuff that would be good to get done in this PR.

**Comment by str4d** [comment]
@MaulingMonkey thanks for bringing this PR across! I didn't create it here originally because at the time this repo looked like a read-only mirror. I'll open my subsequent PRs here instead 🙂

**Pull request merged**